tests/all2all/alltoallconfig_test.go

```diff
@@ -0,0 +1,118 @@
+package all2all
+
+import (
+	"flag"
+	"fmt"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/relab/gorums"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+var replicaCount = flag.Int("replicas", 20, "number of replicas to create all-to-all communication")
+
+func TestAllToAllConfiguration(t *testing.T) {
+	replicas, err := createReplicas(*replicaCount)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		for _, replica := range replicas {
+			replica.stopServer()
+		}
+	}()
+	nodeMap := make(map[string]uint32)
+	for _, replica := range replicas {
+		nodeMap[replica.address] = replica.id
+	}
+	for _, replica := range replicas {
+		replica.createConfiguration(nodeMap)
```

Comment:
The error returned by this method is not checked.

Reply:
Interestingly, this seems to bring back the bug we are fighting with here. If we run the code with:
```sh
$ go test -v -run TestAllToAllConfiguration -replicas 10
=== RUN   TestAllToAllConfiguration
--- PASS: TestAllToAllConfiguration (1.06s)
```
vs 
```sh
go test -v -run TestAllToAllConfiguration -replicas 11
=== RUN   TestAllToAllConfiguration
    alltoallconfig_test.go:33: could not create configuration: connection failed for addr: 127.0.0.1:65204: dialing node failed: context deadline exceeded
--- FAIL: TestAllToAllConfiguration (1.16s)
```
And this is consistently working for up to 10 replicas, but not for 11 or above.

Reply:
I added `TestGrpcDial` to try to understand better what's going on. This version fails consistently with 84 replicas (but not with fewer on my machine that is):
```sh
go test -v -run TestGrpc -replicas 84
=== RUN   TestGrpcDial
    alltoallconfig_test.go:133: context deadline exceeded
--- FAIL: TestGrpcDial (2.05s)
```

Reply:
I can't reproduce this on my machine

```
❯ ./all2all.test -test.v -test.run="TestGrpcDial" -replicas=10000
=== RUN   TestGrpcDial
--- PASS: TestGrpcDial (4.84s)
PASS
```

It only fails for me when the kernel runs out of ports 😅 

```
❯ ./all2all.test -test.v -test.run="TestGrpcDial" -replicas=50000 
=== RUN   TestGrpcDial
    alltoallconfig_test.go:123: listen tcp 127.0.0.1:0: bind: address already in use
--- FAIL: TestGrpcDial (9.77s)
FAIL
```